pkg/controller/v1beta1/inferenceservice/reconcilers/hpa/hpa_reconciler.go

```diff
@@ -44,30 +46,40 @@ type HPAReconciler struct {
 func NewHPAReconciler(client client.Client,
 	scheme *runtime.Scheme,
 	componentMeta metav1.ObjectMeta,
-	componentExt *v1beta1.ComponentExtensionSpec) *HPAReconciler {
+	componentExt *v1beta1.ComponentExtensionSpec) (*HPAReconciler, error) {
+	hpa, err := createHPA(componentMeta, componentExt)
+	if err != nil {
+		return nil, err
+	}
 	return &HPAReconciler{
 		client:       client,
 		scheme:       scheme,
-		HPA:          createHPA(componentMeta, componentExt),
+		HPA:          hpa,
 		componentExt: componentExt,
-	}
+	}, nil
 }
 
-func getHPAMetrics(metadata metav1.ObjectMeta, componentExt *v1beta1.ComponentExtensionSpec) []autoscalingv2.MetricSpec {
+func getHPAMetrics(metadata metav1.ObjectMeta, componentExt *v1beta1.ComponentExtensionSpec) ([]autoscalingv2.MetricSpec, error) {
 	var metrics []autoscalingv2.MetricSpec
 	var utilization int32
+	var err error
 	annotations := metadata.Annotations
 	resourceName := corev1.ResourceCPU
 
 	if value, ok := annotations[constants.TargetUtilizationPercentage]; ok {
-		utilizationInt, _ := strconv.Atoi(value)
-		utilization = int32(utilizationInt) // #nosec G109
+		utilization, err = utils.StringToInt32(value)
+		if err != nil {
+			return metrics, err
+		}
 	} else {
 		utilization = constants.DefaultCPUUtilization
 	}
 
 	if componentExt.ScaleTarget != nil {
-		utilization = int32(*componentExt.ScaleTarget)
+		utilization, err = utils.StringToInt32(strconv.Itoa(*componentExt.ScaleTarget))
```

Comment:
scaleTarget is defined as int, it is better to change the API type to int32.

Reply:
Done